pkg/osutils/filesystem/filesystem.go

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2022 Aspect Build Systems, Inc.
+ *
+ * Licensed under the aspect.build Commercial License (the "License");
+ * you may not use this file except in compliance with the License.
+ * Full License text is in the LICENSE file included in the root of this repository.
+ */
+
+package filesystem
+
+import (
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"time"
+)
+
+func timeSince(t time.Time) time.Duration {
+	return time.Since(t)
+}
+
+func timeUnix(sec int64, nsec int64) time.Time {
+	return time.Unix(sec, nsec)
+}
+
+func ioutilTempDir(dir string, pattern string) (name string, err error) {
+	return ioutil.TempDir(dir, pattern)
+}
+
+func osMkdirAll(path string, perm fs.FileMode) error {
+	return os.MkdirAll(path, perm)
+}
+
+func osRename(oldpath string, newpath string) error {
+	return os.Rename(oldpath, newpath)
+}
+
+type ExecCmdRunner interface {
+	Output() ([]byte, error)
+}
+
+func osExecCommand(name string, arg ...string) ExecCmdRunner {
+	return exec.Command(name, arg...)
+}
+
+func osStat(name string) (fs.FileInfo, error) {
+	return os.Stat(name)
+}
+
+type Filesystem struct {
```

Comment:
Could you comment on the exported types based on the godoc good practices, please?